packages/synapse-react-client/src/components/ChangePassword/useChangePasswordFormState.tsx

```diff
@@ -44,8 +46,9 @@ export default function useChangePasswordFormState(
         // e.g. if the user enters the wrong 2FA code
         setTwoFactorAuthErrorResponse(maybeTwoFactorResponse)
       } else {
+        setSuccessfullyChangedPassword(true)
```

Comment:
We have to maintain this in a separate state variable, because `isSuccess` returned by the mutation will be true when the server issues a 2FA challenge